### PR TITLE
network: Cleaning up `vmispec`

### PR DESCRIPTION
**Pull request by EdDev**
**What this PR does / why we need it**:
Refactor the `vmispec` and virt-controller dynamic interfaces calculation.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by EdDev** [comment]
change: Rebase

File: pkg/network/vmispec/interface.go [hunk 1]
```diff
@@ -128,3 +118,13 @@ func LookupInterfaceByName(ifaces []v1.Interface, name string) *v1.Interface {
 	}
 	return nil
 }
+
+func IndexInterfaceStatusByName(interfaces []v1.VirtualMachineInstanceNetworkInterface, p func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool) map[string]v1.VirtualMachineInstanceNetworkInterface {
```
Comment (dhiller) [inline pkg/network/vmispec/interface.go:122]: I would suggest to rename `p` into something more descriptive, probably `doesMatch` or `isMatching`. WDYT?

File: pkg/network/vmispec/hotplug_test.go [hunk 2]
```diff
@@ -40,49 +40,6 @@ var _ = Describe("utilitary funcs to identify attachments to hotplug", func() {
 		networkName    = "n1"
 	)
 
-	DescribeTable("NetworksToHotplug", func(vmi *v1.VirtualMachineInstance, networksToHotplug ...v1.Network) {
```
Comment (dhiller) [inline pkg/network/vmispec/hotplug_test.go:43]: Are these tests safe to be removed?

**Review by ormergi — APPROVED** [review]
Looks good, thanks for the PR

**Pull request merged**